### ISSUE TITLE
[regression] hevc encoding 

### ISSUE THREAD

**Issue by xhaihao** [Intel-Media-SDK/MediaSDK#1894]
Recently MSDK broke the pre-check for gst-msdk (https://github.com/intel-media-ci/gst-plugins-bad/pull/52), the result of git bisect shows 9120f46035bca1899ea586856bca7b3a97741bb9 is the first bad commit. 

You may check  https://media-ci.ostc.intel.com/build/365163/overview for the error case

**Comment by dmitryermilov** [comment]
App may sets property qpp=14 but don’t set qpi or qpb properties.  Thus, app will send QPP=14, QPI=0 and QPB=0 to MSDK.  Before the MSDK change , MSDK will translate/clamp par.mfx.QPP=14, par.mfx.QPI=1 and par.mfx.QPB=1.  After the MSDK change, MSDK will set par.mfx.QPP=0, par.mfx.QPI=0 and par.mfx.QPB=0.

**Comment by dmitryermilov** [comment]
@saosipov , @vilichev , @nkurina , I'm reverting this commit. Please provide a another fix for the original issue which this commit was supported to fix

**Comment by uartie** [comment]
On KBL, the regression started with commit 8f5f681b2813.  Here are some details to reproduce:

1. Generate reference bitstream:
```
$ gst-launch-1.0 -vf videotestsrc num-buffers=300 \
  ! video/x-raw,format=NV12,width=352,height=288 ! filesink location=input.yuv
```

2. Do encode, decode and psnr with previous good commit (006b76860993):
```
$ gst-launch-1.0 -vf filesrc location=input.yuv num-buffers=300 \
  ! rawvideoparse format=nv12 width=352 height=288 \
  ! msdkh265enc rate-control=cqp gop-size=30 qpp=14 target-usage=4 \
    num-slices=1 b-frames=0 low-power=false \
  ! video/x-h265,profile=main ! h265parse ! filesink location=006b76860993.h265
$ gst-launch-1.0 -vf filesrc location=006b76860993.h265 ! h265parse ! msdkh265dec \
  ! checksumsink2 frame-checksum=false dump-output=true \
    dump-location=006b76860993.yuv qos=false
$ metrics_calc_lite -i1 input.yuv -i2 006b76860993.yuv -w 352 -h 288 \
  -nopfm -st nv12p psnr all

  <avg_metric=Y-PSNR> 57.32558</avg_metric>
  <avg_metric=U-PSNR> 89.15926</avg_metric>
  <avg_metric=V-PSNR> 86.14896</avg_metric>
  <avg_metric=PSNR> 59.08436</avg_metric>
```

3. Do encode, decode and psnr with FBC commit (8f5f681b2813):
```
$ gst-launch-1.0 -vf filesrc location=input.yuv num-buffers=300 \
  ! rawvideoparse format=nv12 width=352 height=288 \
  ! msdkh265enc rate-control=cqp gop-size=30 qpp=14 target-usage=4 \
    num-slices=1 b-frames=0 low-power=false \
  ! video/x-h265,profile=main ! h265parse ! filesink location=8f5f681b2813.h265
$ gst-launch-1.0 -vf filesrc location=8f5f681b2813.h265 ! h265parse ! msdkh265dec \
  ! checksumsink2 frame-checksum=false dump-output=true \
    dump-location=8f5f681b2813.yuv qos=false
$ metrics_calc_lite -i1 input.yuv -i2 8f5f681b2813.yuv -w 352 -h 288 \
  -nopfm -st nv12p psnr all
  
  <avg_metric=Y-PSNR> 43.30027</avg_metric>
  <avg_metric=U-PSNR> 50.44776</avg_metric>
  <avg_metric=V-PSNR> 49.65481</avg_metric>
  <avg_metric=PSNR> 44.62328</avg_metric>
```

I have attached the files for your reference:

[files.tar.gz](https://github.com/Intel-Media-SDK/MediaSDK/files/4013226/files.tar.gz)

**Comment by uartie** [comment]
For further context, gst-msdk sends QPP=14, QPI=0 and QPB=0 to MSDK.  The problem is that MSDK processes these values differently before and after commit 8f5f681b2813.

**Comment by uartie** [comment]
...I assume @xhaihao is using a newer platform which takes the other code path for handling the QP values.  Hence, did not regress in the same way until 9120f46 (as reported).  But I expect the outcome can be reproduced on the newer platform in the same way as a described previously for my KBL result.

**Comment by xhaihao** [comment]
@uartie @dmitryermilov I just compared the md5 value of the encoding output when running git bisect, and found the the md5 value was changed since 9120f46035bca1899ea586856bca7b3a97741bb9, it is possible the md5 value is changed but the quality is still on-par with before. Now I think the quality drop was caused by 8f5f681b2813ed54666336b0385eb6303a6b256c

**Comment by saosipov** [comment]
Hi guys and Happy New Year!
I'm familiar with this commit. It fixes potential confusion from these params: QPP=14, QPI=0 and QPB=0
Are they used intentionally? What is the intention? QP = 0 means some default value, it it not real 0. Real 0 is not supported in MSDK historically. So the logic is the following. One can set all QPs != 0 or if he wants some automation he can set QPI only and QPP will be QPI+2 and QPB will be QPP+2. If QPI is not set (the same is your QPI=0) it will be set to default 26.
So your case (QPP=14, QPI=0 and QPB=0) looks confusing. What are your expectation? What QPI and QPB are expected? MSDK can't calculate QPI and QPB from QPP.
-Sergey

**Comment by uartie** [comment]
When app sends QPP=14, QPI=0 and QPB=0, the expectation is that MSDK will use QPP=14, QPI=default and QPB=default.

**Comment by uartie** [comment]
> When app sends QPP=14, QPI=0 and QPB=0, the expectation is that MSDK will use QPP=14, QPI=default and QPB=default.

Now, MSDK forces QPP=0, QPI=0 and QPB=0 during check function (which get's interpreted as defaults later, AFAIU).  Before, MSDK check function used to force QPI=1 and QPB=1 (and leave QPP=14) in check function.

**Comment by dvrogozh** [comment]
@xhaihao, @uartie : does #1930 work for you? can you try it out?

**Comment by xhaihao** [comment]
It works for me.

**Comment by uartie** [comment]
WFM

**Comment by dmitryermilov** [comment]
Fixed by #1930
